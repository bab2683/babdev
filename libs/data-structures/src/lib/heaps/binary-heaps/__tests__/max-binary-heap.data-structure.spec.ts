import { MaxBinaryHeap } from '../max-binary-heap.data-structure';

describe('MaxBinaryHeap', () => {
  let heap: MaxBinaryHeap;

  describe('insert', () => {
    beforeEach(() => {
      heap = new MaxBinaryHeap([41, 39, 33, 18, 27, 12]);
    });

    it('should insert 55 and place it as first element', () => {
      heap.insert(55);

      expect(heap.values[0]).toEqual(55);
      expect(heap.values).toMatchSnapshot();
    });
  });

  describe('extractMax', () => {
    it('should return 41 and place the new max as 39', () => {
      heap = new MaxBinaryHeap([41, 39, 33, 18, 27, 12]);
      const result = heap.extractMax();

      expect(result).toEqual(41);
      expect(heap.values[0]).toEqual(39);
      expect(heap.values).toMatchSnapshot();
    });

    it('should return 87 and place the new max as 84', () => {
      heap = new MaxBinaryHeap();
      heap.insert(24);
      heap.insert(35);
      heap.insert(87);
      heap.insert(15);
      heap.insert(23);
      heap.insert(84);
      heap.insert(1);

      const result = heap.extractMax();

      expect(result).toEqual(87);
      expect(heap.values[0]).toEqual(84);
      expect(heap.values).toMatchSnapshot();
    });

    it('should return 35 and place the new max as 24', () => {
      heap = new MaxBinaryHeap();
      heap.insert(15);
      heap.insert(35);
      heap.insert(24);
      heap.insert(19);
      heap.insert(1);
      heap.insert(20);

      const result = heap.extractMax();

      expect(result).toEqual(35);
      expect(heap.values[0]).toEqual(24);
      expect(heap.values).toMatchSnapshot();
    });

    it('should return 87 and place the new max as 84', () => {
      heap = new MaxBinaryHeap();
      heap.insert(15);
      heap.insert(49);
      heap.insert(24);
      heap.insert(19);
      heap.insert(1);
      heap.insert(20);
      heap.insert(13);
      heap.insert(21);

      const result = heap.extractMax();

      expect(result).toEqual(49);
      expect(heap.values[0]).toEqual(24);
      expect(heap.values).toMatchSnapshot();
    });
  });
});
