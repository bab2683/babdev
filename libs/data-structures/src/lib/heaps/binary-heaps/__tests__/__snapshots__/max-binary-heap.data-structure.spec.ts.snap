// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaxBinaryHeap extractMax should return 35 and place the new max as 24 1`] = `
Array [
  24,
  19,
  20,
  15,
  1,
]
`;

exports[`MaxBinaryHeap extractMax should return 41 and place the new max as 39 1`] = `
Array [
  39,
  27,
  33,
  18,
  12,
]
`;

exports[`MaxBinaryHeap extractMax should return 87 and place the new max as 84 1`] = `
Array [
  84,
  24,
  35,
  15,
  23,
  1,
]
`;

exports[`MaxBinaryHeap extractMax should return 87 and place the new max as 84 2`] = `
Array [
  24,
  21,
  20,
  19,
  1,
  15,
  13,
]
`;

exports[`MaxBinaryHeap insert should insert 55 and place it as first element 1`] = `
Array [
  55,
  39,
  41,
  18,
  27,
  12,
  33,
]
`;
